src/devices_models/devices/common/objective_function/market_bid.jl

```diff
@@ -340,22 +341,61 @@ function _add_pwl_term!(
     ::U,
     ::V,
 ) where {T <: PSY.Component, U <: VariableType, V <: AbstractDeviceFormulation}
-    name = PSY.get_name(component)
-    incremental_offer_curve = PSY.get_incremental_offer_curves(cost_function)
-    value_curve = PSY.get_value_curve(incremental_offer_curve)
-    cost_component = PSY.get_function_data(value_curve)
-    base_power = get_base_power(container)
-    device_base_power = PSY.get_base_power(component)
-    power_units = PSY.get_power_units(incremental_offer_curve)
+    return _add_pwl_term_helper!(
+        container,
+        component,
+        cost_function,
+        Incremental,
+        U(),
+        V())
+end
 
-    data = get_piecewise_incrementalcurve_per_system_unit(
-        cost_component,
-        power_units,
-        base_power,
-        device_base_power,
-    )
+"""
+Add PWL cost terms for data coming from the MarketBidCost
+with a fixed decremental offer curve
+"""
+function _add_pwl_term_decremental!(container::OptimizationContainer,
+    component::T,
+    cost_function::PSY.MarketBidCost,
+    ::PSY.CostCurve{PSY.PiecewiseIncrementalCurve},
+    ::U,
+    ::V) where {T <: PSY.Component, U <: VariableType,
+    V <: AbstractDeviceFormulation}
+    return _add_pwl_term_helper!(
+        container,
+        component,
+        cost_function,
+        Decremental,
+        U(),
+        V())
+end
 
-    cost_is_convex = PSY.is_convex(data)
+function _add_pwl_term_helper!(
+    container::OptimizationContainer,
+    component::T,
+    cost_function::PSY.MarketBidCost,
+    offer_type::MarketBidOfferType,
+    ::U,
+    ::V) where {T <: PSY.Component, U <: VariableType,
+    V <: AbstractDeviceFormulation}
+    if offer_type == Incremental
+        cost_data = PSY.get_incremental_offer_curves(cost_function)
+        data = _get_pwl_data(container, component, cost_data)
+        cost_is_convex = PSY.is_convex(data)
+        offer_variable = PieceWiseLinearBlockOffer
+        constraint_variable = PieceWiseLinearBlockOfferConstraint
+        _get_pwl_cost_expression_function = _get_pwl_cost_expression
+    elseif offer_type == Decremental
```

Comment:
Maybe do the `Incremental`/`Decremental`-specific stuff in the incremental/decremental-specific functions that call this helper function and pass the data into here as arguments?